tutorials/models_mapping/plot_1_tensor_models.py

```diff
@@ -0,0 +1,333 @@
+
+
+"""
+Tensor Meshes
+=============
+
+Here we demonstrate various ways that models can be defined and mapped to
+tensor meshes. Some things we consider are:
+
+    - Surface topography
+    - Adding structures of various shape to the model
+    - Parameterized models
+    - Models with 2 or more physical properties
+    
+
+"""
+
+#########################################################################
+# Import modules
+# --------------
+#
+
+from SimPEG import Mesh
+from SimPEG import Utils
+from SimPEG import Maps
+import numpy as np
+import matplotlib.pyplot as plt
+
+# sphinx_gallery_thumbnail_number = 3
+
+#############################################
+# Defining the mesh
+# -----------------
+#
+# Here, we create the tensor mesh that will be used for all examples.
+#
+
+
+def makeExampleMesh():
+
+    dh = 5.
+    hx = [(dh, 5, -1.3), (dh, 20), (dh, 5, 1.3)]
+    hy = [(dh, 5, -1.3), (dh, 20), (dh, 5, 1.3)]
+    hz = [(dh, 5, -1.3), (dh, 20), (dh, 5, 1.3)]
+    mesh = Mesh.TensorMesh([hx, hy, hz], 'CCC')
+
+    return mesh
+
+
+#############################################
+# Halfspace model with topography at z = 0
+# ----------------------------------------
+#
+# In this example we generate a half-space model. Since air cells remain
+# constant during geophysical inversion, the number of model values we define
+# should be equal to the number of cells lying below the surface. Here, we
+# define the model (*mod* ) as well as the mapping (*modMap* ) that goes from
+# the model-space to the entire mesh.
+#
+
+mesh = makeExampleMesh()
+
+m_val = 100.
```

Comment:
`m_halfspace` (`val` is a bit non-descript)

Reply:
update unit names